### DISCUSSION TITLE
Ошибка при компиляции.

### DISCUSSION THREAD

**Discussion by Alexrd6** [vvip-68/GyverPanelWiFi#211] [Задавайте вопросы] [answered]
Здравствуйте! Версия прошивки GyverPanelWiFi_v1.12.ino. При компиляции выдает ошибку:

a_def_hard.h:292:188: fatal error: a_def_devc.h: No such file or directory
 #include "a_def_devc.h"       // Если здесь ошибка - смотри комментарий к определению A_DEF_DEVC выше в начале этого файла
                                                                                                                                                                                            ^
compilation terminated.
exit status 1
a_def_devc.h: No such file or directory

Что не так? Спасибо!

**Comment by vvip-68** [comment, marked as answer]
Комментарий читать пробовали?

**Comment by Alexrd6** [comment]
Я новичок в этом, но очень хочу повторить ваш проект. Подскажите куда копать? ПЛЗ!

**Reply by vvip-68** [reply]
#include "a_def_devc.h" // Если здесь ошибка - смотри комментарий к определению A_DEF_DEVC выше в начале этого файла

**Comment by Alexrd6** [comment]
Ок СПАСИБО! Разобрался! Сорри, сам ступил! Ещё раз СПАСИБО! Всё пошло!!!!!!

**Comment by faktor2** [comment]
При компеляции выдаёт ошибку
![Screenshot_7](https://user-images.githubusercontent.com/116493700/197391040-cdf9be3d-a10a-4ba8-a3b3-bf833ed28209.jpg)
что это может быть?

**Reply by vvip-68** [reply]
Неверная версия ядра системы, неверная версия IDE, испорченный исходный код приложения...
Ну, еще что-нибудь...

**Comment by Vovilen** [comment]
Здравствуйте.
Не компилится даже если просто нажать кнопку Компилировать неговоря уже про залить в контроллер.
В менеджере плат выбрал 2.7.4 NodeMcu ESP12E module

DEVICE_VARIANT 0
1. Контроллер ESP8266 LoLin ver. 0.1 NodeMcu v3
2. Контроллер ESP8266 WeMos D1 Mini PRO V3.0.0
Прошивка 1.12
FastLED 3.3.0

C:\Users\Dell\Documents\Arduino\libraries\ArduinoJson-6.x\src/ArduinoJson/MsgPack/MsgPackDeserializer.hpp: In member function 'ArduinoJson6194_F1::DeserializationError::Code ArduinoJson6194_F1::MsgPackDeserializer<TReader, TStringStorage>::readObject(ArduinoJson6194_F1::VariantData*, size_t, TFilter, ArduinoJson6194_F1::NestingLimit)':
C:\Users\Dell\Documents\Arduino\libraries\ArduinoJson-6.x\src/ArduinoJson/MsgPack/MsgPackDeserializer.hpp:473:35: error: expected unqualified-id before string constant
       String key = _stringStorage.str();
                                   ^
exit status 1
Ошибка компиляции для платы NodeMCU 1.0 (ESP-12E Module).

**Reply by vvip-68** [reply]
Возьмите библиотеку ArduinoJson из папки проекта.

**Comment by whitevv** [comment]
Здравствуйте!
На прошивке 1.12, всё отлично работает!
При компиляции 1.13, ошибка 
"index_map:121:2: error: #error Нет определения карты индексов для заданных WIDTH и HEIGHT
exit status 1
#error Нет определения карты индексов для заданных WIDTH и HEIGHT"

...
#define DEVICE_VARIANT 0
...

#if (DEVICE_VARIANT == 0)

#define WIDTH 15                // ширина одного сегмента матрицы для MATRIX_TYPE == 0 или MATRIX_TYPE == 1                                       | для MATRIX_TYPE == 2 ПОЛНАЯ ШИРИНА матрицы
#define HEIGHT 20               // высота одного сегмента матрицы для MATRIX_TYPE == 0 или MATRIX_TYPE == 1                                       | для MATRIX_TYPE == 2 ПОЛНАЯ ВЫСОТА матрицы 
#define MATRIX_TYPE 0           // тип соединения диодов в сегменте матрицы: 0 - зигзаг, 1 - параллельная, 2 - использовать карту индексов
#define MATRIX_INDEX 0          // только для MATRIX_TYPE == 2 - место расположения массива индексов 0 - в сектче, 1 - в файле  
#define CONNECTION_ANGLE 0      // угол подключения диодов в сегменте: 0 - левый нижний, 1 - левый верхний, 2 - правый верхний, 3 - правый нижний | для MATRIX_TYPE == 2 - не используется
#define STRIP_DIRECTION 1       // направление ленты из угла сегмента: 0 - вправо, 1 - вверх, 2 - влево, 3 - вниз                                 | для MATRIX_TYPE == 2 - не используется

#define META_MATRIX_WIDTH 1     // количество сегментов в ширину сборной матрицы                                                                  | для MATRIX_TYPE == 2 - не используется
#define META_MATRIX_HEIGHT 1    // количество сегментов в высоту сборной матрицы                                                                  | для MATRIX_TYPE == 2 - не используется
#define META_MATRIX_TYPE 0      // тип сборной матрицы: 0 - зигзаг, 1 - параллельная                                                              | для MATRIX_TYPE == 2 - не используется
#define META_MATRIX_ANGLE 0     // угол 1-го сегмента сборной матрицы: 0 - левый нижний, 1 - левый верхний, 2 - правый верхний, 3 - правый нижний | для MATRIX_TYPE == 2 - не используется
#define META_MATRIX_DIRECTION 0 // направление следующих сегментов сборной матрицы из угла: 0 - вправо, 1 - вверх, 2 - влево, 3 - вниз            | для MATRIX_TYPE == 2 - не используется

#define DEVICE_TYPE 1           // Использование матрицы: 0 - свернута в трубу для лампы; 1 - плоская матрица в рамке   
#define BUTTON_TYPE 1           // Тип кнопки: 0 - сенсорная (touch button); 1 - обычная кнопка на замыкание при нажатии (или если кнопки нет вообще)  

#define USE_SD 0                // поставьте 0, если у вас нет SD-card модуля
#define USE_TM1637 0            // поставьте 1, если используется дополнительный индикатор TM1637, 0 если индикатора нет
#define USE_POWER 0             // 1 - использовать отключение питания матрицы через MOSFET / реле; 0 - не использовать 
#define USE_MQTT 1              // 1 - использовать управление по MQTT-каналу; 0 - не использовать 
#define USE_WEATHER 1           // 1 - использовать получение информации о текущей погоде; 0 - не использовать 
#define USE_E131 0              // 1 - использовать протокол E1.31 для синхронизации устройств или приема внешнего потока на матрицу с Jinx! или другого аналогичного ПО
#define BIG_FONT 0              // 0 - шрифт 5x8, 1 - шрифт  10x16; 2 - шрифт 8x13

**Reply by vvip-68** [reply]
Ну да, косячок-с... Поставьте в 
```
#define MATRIX_INDEX 1
```

**Reply by vvip-68** [reply]
А я вот вижу в этом файле что `#define MATRIX_INDEX 1`
Чтобы не было такой ошибки. А у вас в тексте выше - там 0.
Зачем исправили?
![image](https://user-images.githubusercontent.com/39353139/204155069-7f4e5272-e166-4eed-8621-bbaeeedff468.png)

**Reply by whitevv** [reply]
Спасибо помогло

**Comment by mihey2011** [comment]
Добрый вечер, мучаюсь с компиляцией скетча, до загрузки даже дело не доходит, выскакивает ошибка: control reaches end of non-void function [-Werror=return-type]
Третий день мучаюсь, в чем может быть дело?
![Скриншот 28-11-2022 200146](https://user-images.githubusercontent.com/119345407/204337775-4cbf2d77-f0ce-4148-9945-a4577b60f53d.jpg)

**Reply by vvip-68** [reply]
Без полной информации об ошибке ( а не того огрызка лога, что на картинке оранжевым) вам вряд ли кто-нибудь скажет что-то осмысленное... Возможно неподходящие версии библиотек установлены. Или версия ядра не та.

**Comment by mihey2011** [comment]
Почистил библиотеки от мусора, и все скомпилировалось, но теперь возникает другая ошибка, но уже при загрузке в плату:
```
In file included from c:\Users\mihey\Documents\Arduino\libraries\FastLED\src/FastLED.h:67,
                 from D:\Миша\Arduino\GyverPanelWiFi-master (1)\GyverPanelWiFi-master\firmware\GyverPanelWiFi_v1.13\a_def_hard.h:391,
                 from D:\Миша\Arduino\GyverPanelWiFi-master (1)\GyverPanelWiFi-master\firmware\GyverPanelWiFi_v1.13\GyverPanelWiFi_v1.13.ino:77:
c:\Users\mihey\Documents\Arduino\libraries\FastLED\src/fastspi.h:145:23: note: '#pragma message: No hardware SPI pins defined.  All SPI access will default to bitbanged output'
  145 | #      pragma message "No hardware SPI pins defined.  All SPI access will default to bitbanged output"
      |                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Executable segment sizes:

ICACHE : 32768           - flash instruction cache 

IROM   : 561500          - code in flash         (default or ICACHE_FLASH_ATTR) 

IRAM   : 28960   / 32768 - code in IRAM          (IRAM_ATTR, ISRs...) 

DATA   : 1848  )         - initialized variables (global, static) in RAM/HEAP 

RODATA : 5352  ) / 81920 - constants             (global, static) in RAM/HEAP 

BSS    : 35192 )         - zeroed variables      (global, static) in RAM/HEAP 

Скетч использует 597660 байт (57%) памяти устройства. Всего доступно 1044464 байт.
Глобальные переменные используют 42392 байт (51%) динамической памяти, оставляя 39528 байт для локальных переменных. Максимум: 81920 байт.
esptool.py v3.0
Serial port COM9
Connecting........_____....._____....._____....._____....._____....._____....._____
Traceback (most recent call last):
  File "C:\Users\mihey\AppData\Local\Arduino15\packages\esp8266\hardware\esp8266\3.0.2/tools/upload.py", line 66, in <module>
    esptool.main(cmdline)
  File "C:/Users/mihey/AppData/Local/Arduino15/packages/esp8266/hardware/esp8266/3.0.2/tools/esptool\esptool.py", line 3552, in main
    esp.connect(args.before, args.connect_attempts)
  File "C:/Users/mihey/AppData/Local/Arduino15/packages/esp8266/hardware/esp8266/3.0.2/tools/esptool\esptool.py", line 529, in connect
    raise FatalError('Failed to connect to %s: %s' % (self.CHIP_NAME, last_error))
esptool.FatalError: Failed to connect to ESP8266: Timed out waiting for packet header
Failed uploading: uploading error: exit status 1
```

**Reply by vvip-68** [reply]
1. Забыли вставить USB-шнур в плату или в компьютер
2. Неисправна плата - заменить
3. Неисправен USB-шнур - заменить
4. Неисправен USB разъем компьютера - отремонтировать
5. Шнур включен в порт USB3, а плата понимает только USB2 и есть какая-то несовместимость - найти другой разъем USB на компьютере
6. Используется не Wemos di mini, а какой-нибудь Node MCU с двумя кнопками - BOOT и RESET. Нажать кнопки одновременно, затем отпустить сначала Reset, затем Boot. Или наоборот - не помню...
7. Менее вероятно - не выбран правильный порт, порт "отвалился" и нужно перезагрузить компьютер, не установлены нужные драйвера на USB для того чипа UART который установлен на вашей плате микроконтроллера.

**Comment by mihey2011** [comment]
Все перепробовал, ничего не помогло, заменил плату. Вроде стало загружаться, но диоды молчат. Высвечивается следующее: 
```
In file included from c:\Users\mihey\Documents\Arduino\libraries\FastLED\src/FastLED.h:67,
                 from D:\Миша\Arduino\GyverPanelWiFi-master (1)\GyverPanelWiFi-master\firmware\GyverPanelWiFi_v1.13\a_def_hard.h:391,
                 from D:\Миша\Arduino\GyverPanelWiFi-master (1)\GyverPanelWiFi-master\firmware\GyverPanelWiFi_v1.13\GyverPanelWiFi_v1.13.ino:77:
c:\Users\mihey\Documents\Arduino\libraries\FastLED\src/fastspi.h:145:23: note: '#pragma message: No hardware SPI pins defined.  All SPI access will default to bitbanged output'
  145 | #      pragma message "No hardware SPI pins defined.  All SPI access will default to bitbanged output"
      |                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Executable segment sizes:

ICACHE : 32768           - flash instruction cache 

IROM   : 529060          - code in flash         (default or ICACHE_FLASH_ATTR) 

IRAM   : 27984   / 32768 - code in IRAM          (IRAM_ATTR, ISRs...) 

DATA   : 1740  )         - initialized variables (global, static) in RAM/HEAP 

RODATA : 5112  ) / 81920 - constants             (global, static) in RAM/HEAP 

BSS    : 33472 )         - zeroed variables      (global, static) in RAM/HEAP 

Скетч использует 563896 байт (53%) памяти устройства. Всего доступно 1044464 байт.
Глобальные переменные используют 40324 байт (49%) динамической памяти, оставляя 41596 байт для локальных переменных. Максимум: 81920 байт.
esptool.py v3.0
Serial port COM9
Connecting....
Chip is ESP8266EX
Features: WiFi
Crystal is 26MHz
MAC: 34:94:54:61:be:93
Uploading stub...
Running stub...
Stub running...
Configuring flash size...
Auto-detected Flash size: 4MB
Compressed 568048 bytes to 389969...
Writing at 0x00000000... (4 %)
Writing at 0x00004000... (8 %)
Writing at 0x00008000... (12 %)
Writing at 0x0000c000... (16 %)
Writing at 0x00010000... (20 %)
Writing at 0x00014000... (25 %)
Writing at 0x00018000... (29 %)
Writing at 0x0001c000... (33 %)
Writing at 0x00020000... (37 %)
Writing at 0x00024000... (41 %)
Writing at 0x00028000... (45 %)
Writing at 0x0002c000... (50 %)
Writing at 0x00030000... (54 %)
Writing at 0x00034000... (58 %)
Writing at 0x00038000... (62 %)
Writing at 0x0003c000... (66 %)
Writing at 0x00040000... (70 %)
Writing at 0x00044000... (75 %)
Writing at 0x00048000... (79 %)
Writing at 0x0004c000... (83 %)
Writing at 0x00050000... (87 %)
Writing at 0x00054000... (91 %)
Writing at 0x00058000... (95 %)
Writing at 0x0005c000... (100 %)
Wrote 568048 bytes (389969 compressed) at 0x00000000 in 36.4 seconds (effective 124.9 kbit/s)...
Hash of data verified.

Leaving...
Hard resetting via RTS pin...
```

**Comment by vvip-68** [comment]
И что не так? Скетч скомпилирован без ошибок, загружен на плату, запустился и начал работать.
Диоды могут не светиться:
- неправильно подключена матрица (не к тому пину),
- не подано питание на матрицу
- матрица просто сгоревшая и не работает.
- прошивка находится в состоянии "Выключено" и ее надо включить с кнопки или с приложения

**Comment by lentiai** [comment]
Уважаемые форумчане, Помогите разобраться. Все делаю строго по инструкции, при пробной компиляции выдает --
cc1plus.exe: fatal error: C:\Users\ËóÑ\AppData\Local\Arduino15\packages\esp8266\hardware\esp8266\3.1.1\cores\esp8266\CommonHFile.h: No such file or directory
compilation terminated
 Папка существует, правда с файлом нулевого размера. Перепробовал все! Куда копать?

**Reply by vvip-68** [reply]
Ядро 3.1.1 не поддерживается данным проектом. Ставьте рекомендованное ядро 2.7.4

**Comment by lentiai** [comment]
Все получилось, Спасибо.

**Comment by lentiai** [comment]
Извините, образовался еще вопрос, после настроек сети в PanelSetupWiFi, замедлилась работа всей системы ( каждое действие через 5-6 секунд). И перепрошивка ни чего не дала. Как сбросить внутреннею память до начальных значений?

**Reply by vvip-68** [reply]
Залить прошивку с включенной опцией в меню "Инструменты" - Flash content - Erase all content

**Reply by vvip-68** [reply]
Тормозит потому что микроконтроллеру нужен доступ в интернет  для получения времени (и погоды), а с вашими настройками у него доступа к интернету нет. Если нет интернета - отключайте получение погоды и синхронизацию времени с часами из интернета

**Comment by lentiai** [comment]
По первому сам нашел.
По второму, не могу разобрался как корректно подключится к домашней сети (там инет есть), Как точка доступа, все работает.

**Reply by vvip-68** [reply]
При подключении как к точке доступа в приложении WiFiPanelSetup на вкладке настройки сети ввести имя сети, пароль доступа и свободный IP адрес в сети вашего роутера. Нажать кнопку "Сохранить".

**Comment by lentiai** [comment]
Так и делаю , если только свободный ip не правильно, Где его точно взять? А порт какой вбивать ?

**Reply by vvip-68** [reply]
Вы не знаете какой IP у вас в сети свободный? Хм... Ну зайдите на Web-страничку настроек роутера, найдите там раздел со списком подключенных устройств, посмотрите какие адреса отсутствуют в этом списке - возьмите один из отсутствующих (свободных) и присвойте его вашей матрице в WiFiPanelSetup. Порт - по умолчанию, какой в скетче прописан 2390 что - ли - не помню. Посмотрите в скетче.

Какой IP у вашего роутера? Какой IP назначаете матрице?

**Reply by vvip-68** [reply]
А где там вообще порт "вбивается"? Я не помню такого...

**Comment by lentiai** [comment]
Спасибо Вам буду разбираться

**Comment by sharik-vova** [comment]
у меня не компилируется под esp8266, под Avr всё работает. Пишет:

/Users/vova/Downloads/GyverMatrixBT-master/firmware/GyverMatrixOS_v1.12/runningText.ino: In function 'uint8_t getFont(uint8_t, uint8_t)':
runningText:96:1: error: control reaches end of non-void function [-Werror=return-type]
   96 | }
      | ^
cc1plus: some warnings being treated as errors
exit status 1
control reaches end of non-void function [-Werror=return-type]

помогите пожалуйста

**Reply by vvip-68** [reply]
Ну, про тот проект лучше всего спрашивать в гитхабе того проекта или в ветке форума, посвященной тому проекту. Тут вряд ли кто-то сможет что-то посоветовать.
Этот проект не поддерживает AVR, тот, вполне возможно - не поддерживает esp8266. Проекты разные, хотя и внешне похожие. Обращайтесь за помощью туда - будет эффективнее, чем писать в первый попавшийся другой проект.

**Reply by Libra27** [reply]
Error compiling for board NodeMCU 1.0 (ESP-12E Module). взял библиотеку ArduinoJson из папки проекта. Прошивка 1.13, пробовал Версии FastLED  - 3.5.0 и 3.4.0 Прошивка 1.12 работает

**Comment by WIT555** [comment]
Вот такая ошибка :
Arduino: 1.8.19 (Windows 7), Плата:"NodeMCU 1.0 (ESP-12E Module), 80 MHz, Flash, Disabled (new aborts on oom), Disabled, All SSL ciphers (most compatible), 32KB cache + 32KB IRAM (balanced), Use pgm_read macros for IRAM/PROGMEM, Mapping defined by Hardware and Sketch, 2, v2 Lower Memory, Disabled, None, Only Sketch, 115200"

. Variables and constants in RAM (global, static), used 39312 / 80192 bytes (49%)

║   SEGMENT  BYTES    DESCRIPTION

╠══ DATA     1700     initialized variables

╠══ RODATA   3876     constants       

╚══ BSS      33736    zeroed variables

. Instruction RAM (IRAM_ATTR, ICACHE_RAM_ATTR), used 60380 / 65536 bytes (92%)

║   SEGMENT  BYTES    DESCRIPTION

╠══ ICACHE   32768    reserved space for flash instruction cache

╚══ IRAM     27612    code in IRAM    

. Code in flash (default, ICACHE_FLASH_ATTR), used 527488 / 1048576 bytes (50%)

║   SEGMENT  BYTES    DESCRIPTION

╚══ IROM     527488   code in flash   



LittleFS Not Defined for NodeMCU 1.0 (ESP-12E Module), 80 MHz, Flash, Disabled (new aborts on oom), Disabled, All SSL ciphers (most compatible), 32KB cache + 32KB IRAM (balanced), Use pgm_read macros for IRAM/PROGMEM, Mapping defined by Hardware and Sketch, 2, v2 Lower Memory, Disabled, None, Only Sketch, 115200

**Reply by vvip-68** [reply]
И в чём ошибка? "Меня пугают оранжевые буквы"?
Из содержимого не очень понятно выделили ли вы место под файловую систему.
Если нет - никакие настройки сохраняться не будут и вам при каждом перезапуске придется все настраивать заново.

**Comment by WIT555** [comment]
Добрый день. Меня пугает то, что матрица не работает. При выделении места под файловую  систему, микроконтроллер не стартует. В мониторе ничего не пишет. При изменении настроек места под файловую систему, скетч не компилируется.

**Reply by vvip-68** [reply]
Микроконтроллер не стартует или скетч не компилируется и не заливается в микроконтроллер? Если нет вывода в лог - значит он либо отключен в настройках, либо действительно микроконтроллер не стартует, либо в мониторе порта выставлена не та скорость, что настроена в скетче (115200). Матрица не светится - или микроконтроллер не стартует, или вход матрицы припаян не к тем пинам, на которые назначен вывод. Судя по куску лога выше - никаких проблем у вас нет. Даже скетч занимает 92% (не 100%!) от доступного пространства. Почему у вас не хватает места - не знаю. Больше ни у кого (в том числе и у меня) таких проблем не возникало. Из предоставленной информации никаких выводов сделать невозможно.

**Reply by vvip-68** [reply]
Вот распределение памяти у меня. Как видите - вообще 96% занято. Всё работает
![image](https://github.com/vvip-68/GyverPanelWiFi/assets/39353139/0fb2e99d-7b92-4b13-ab2a-60924e1742b2)

**Reply by WIT555** [reply]
А есть какие то отдельные настройки микроконтроллера? И если есть, то где их посмотреть и настроить. Может это поможет исправить ситуацию.

**Reply by vvip-68** [reply]
Ну все настройки куда что подключать, какие возможности использовать - в файле a_def_hard.h
А так я даже не знаю какой версии проект вы используете. Судя по тому ГДЕ размещена переписка - это версия 1.13. Но судя по логам компиляции - у вас ядро версии примерно 3.1.2 с которым 1.13 не работает. Если проект 1.14, который работает на ядре 3.1.2 - то после компиляции, загрузки прошивки и файлов в файловую систему - практически всё можно настроить из Web-интерфейса.

Ну и вообще надо понимать какие возможности у вас в прошивке включены (дефайны вида #define USE_XXX) и какие пины микроконтроллера вы используете. Микроконтроллер может вообще не стартовать, если во время включения на GPIO 0,2,15 выставлены неправильные уровни (не то и не туда припаяли).

**Reply by WIT555** [reply]
После Ваших разъяснений микроконтроллер стартовал. Вроде нормально. При включении и редактировании в скетч Вашего примерного файла a_def_devc.h , ругается  на 72 строку так: exit status 1  , unterminated #if . Что это?

**Reply by vvip-68** [reply]
директива условной компиляции - 
```
#if условие
тут то что должно включаться при выполнении условия
#endif
```
должна заканчиваться директивой конца `#endif`
Вероятно вы где-то что-то редактировали и удалили лишний `#endif` таким образом, блок открывается `#if условие`, а вот закрывающего тега нет. Нарушена структура кода.

**Reply by WIT555** [reply]
Еще такой вопрос : кнопка при монтаже обязательна? И если на роутере нет пароля для подключения к сети вайфай, оставлять пустое место в кавычках, при редактировании?

**Reply by vvip-68** [reply]
Кнопка не обязательна. Нужно отключить в a_dev_hard.h  `#USE_BUTTON 0` иначе яркость может самопроизвольно уменьшаться или увеличиваться. Если микроконтроллер не может подключиться к сети , он будет три минуты ждать появления сети, прежде чем создать точку доступа Если есть кнопка - ожидание можно прервать двойным кликом кнопки. Если нет - ждать. Про пустой пароль - не подскажу. Не встречал сетей с пустым паролем, соответственно данный вариант не тестировался даже. Пробуйте.

**Reply by WIT555** [reply]
Спасибо. Всё заработало.

**Comment by fenixmonolith** [comment]
ребята помогите пожалуйста. в кодах не разбираюсь, вроде пару месяцев назад без проблем было тут такое

In file included from c:\Users\abror by light\Documents\Arduino\libraries\FastLED\src/FastLED.h:75,
                 from C:\Users\abror by light\Downloads\GyverMatrixWiFi-master\GyverMatrixWiFi-master\firmware\GyverMatrixOS_v1.13\GyverMatrixOS_v1.13.ino:18:
c:\Users\abror by light\Documents\Arduino\libraries\FastLED\src/fastspi.h:157:23: note: '#pragma message: No hardware SPI pins defined.  All SPI access will default to bitbanged output'
  157 | #      pragma message "No hardware SPI pins defined.  All SPI access will default to bitbanged output"
      |                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
C:\Users\abror by light\Downloads\GyverMatrixWiFi-master\GyverMatrixWiFi-master\firmware\GyverMatrixOS_v1.13\clock.ino: In function 'long unsigned int sendNTPpacket(IPAddress&)':
C:\Users\abror by light\Downloads\GyverMatrixWiFi-master\GyverMatrixWiFi-master\firmware\GyverMatrixOS_v1.13\clock.ino:60:3: error: return-statement with no value, in function returning 'long unsigned int' [-fpermissive]
   60 | }
      |   ^     
C:\Users\abror by light\Downloads\GyverMatrixWiFi-master\GyverMatrixWiFi-master\firmware\GyverMatrixOS_v1.13\runningText.ino: In function 'uint8_t getFont(uint8_t, uint8_t)':
C:\Users\abror by light\Downloads\GyverMatrixWiFi-master\GyverMatrixWiFi-master\firmware\GyverMatrixOS_v1.13\runningText.ino:100:3: error: return-statement with no value, in function returning 'uint8_t' {aka 'unsigned char'} [-fpermissive]
  100 | }
      |   ^     

exit status 1

Compilation error: return-statement with no value, in function returning 'long unsigned int' [-fpermissive]

**Reply by vvip-68** [reply]
GyverMatrixOS - это совсем другой проект. Почему вы спрашиваете про него здесь?
Ну а по делу - по внешнему виду вы используете ядро ESP8266 версии примерно 3.1.2. Конкретно тот проект создавался в то время, когда ядро было примерно версии 2.5.2 и совершенно не рассчитан на новые версии ядра. Используйте версию, рекомендованную автором для этого проекта. Должно также работать под  2.7.4.

**Reply by fenixmonolith** [reply]
Спасибо за быстрый ответ, загулялся в темах((
и сейчас выдает:


Traceback (most recent call last):
  File "C:\Users\abror by light\AppData\Local\Arduino15\packages\esp8266\hardware\esp8266\2.7.4/tools/upload.py", line 65, in <module>
    esptool.main(cmdline)
  File "C:/Users/abror by light/AppData/Local/Arduino15/packages/esp8266/hardware/esp8266/2.7.4/tools/esptool\esptool.py", line 2889, in main
    esp = chip_class(each_port, initial_baud, args.trace)
  File "C:/Users/abror by light/AppData/Local/Arduino15/packages/esp8266/hardware/esp8266/2.7.4/tools/esptool\esptool.py", line 249, in __init__
    self._port.write_timeout = DEFAULT_SERIAL_WRITE_TIMEOUT
  File "C:/Users/abror by light/AppData/Local/Arduino15/packages/esp8266/hardware/esp8266/2.7.4/tools/pyserial\serial\serialutil.py", line 388, in write_timeout
    self._reconfigure_port()
  File "C:/Users/abror by light/AppData/Local/Arduino15/packages/esp8266/hardware/esp8266/2.7.4/tools/pyserial\serial\serialwin32.py", line 222, in _reconfigure_port
    'Original message: {!r}'.format(ctypes.WinError()))
serial.serialutil.SerialException: Cannot configure port, something went wrong. Original message: PermissionError(13, '�������������� � ������� ���������� �� ��������.', None, 31)
Failed uploading: uploading error: exit status 1

**Reply by vvip-68** [reply]
Ну, порт чем-то занят. Кто-то уже открыл его до того как его пытается открыть  загрузчик скетча.
Смотрите что держит его открытым. Просто перезагрузите комп, может поможет. Кабель смените.

**Reply by fenixmonolith** [reply]
обновил драйвера, подключил в другой разьем заработало спасибо вам большое))

**Comment by Hm7284156** [comment]
C:\Users\Хабиб\Desktop\GyverPanelWiFi-master\GyverPanelWiFi-master\firmware\GyverPanelWiFi_v1.13\e131.ino:393:21: error: cannot convert 'CRGB' to 'uint32_t' {aka 'unsigned int'} in assignment
C:\Users\Хабиб\Desktop\GyverPanelWiFi-master\GyverPanelWiFi-master\firmware\GyverPanelWiFi_v1.13\e131.ino:463:21: error: cannot convert 'CRGB' to 'uint32_t' {aka 'unsigned int'} in assignment

exit status 1

Compilation error: cannot convert 'CRGB' to 'uint32_t' {aka 'unsigned int'} in assignment

**Reply by vvip-68** [reply]
Не та версия ядра, не та версия библиотек, не та версия IDE...
В общем - не та версия чего-то...

**Comment by WIT555** [comment]
Добрый день. Как сделать меньше размеры анимации?

**Reply by vvip-68** [reply]
Создать кадры анимации нужного размера, в блоке описателя анимации указать размеры, соответствующие созданным кадрам.

**Reply by WIT555** [reply]
Я про анимацию, которая уже создана, в скетче. Как вариант- "Марио". Он у меня на полгирлянды. Хотелось бы уменьшить

**Comment by Inthestep** [comment]
Здрасте. Ошибка такова:

In file included from C:\Users\myUser\OneDrive\Рабочий стол\GyverPanelWiFi-master\firmware\GyverPanelWiFi_v1.13\a_def_hard.h:404,
                 from C:\Users\myUser\OneDrive\Рабочий стол\GyverPanelWiFi-master\firmware\GyverPanelWiFi_v1.13\GyverPanelWiFi_v1.13.ino:88:
C:\Users\myUser\OneDrive\���������\Arduino\libraries\ArduinoJson-6.x/ArduinoJson.h:5:10: fatal error: src/ArduinoJson.h: No such file or directory
    5 | #include "src/ArduinoJson.h"
      |          ^~~~~~~~~~~~~~~~~~~
compilation terminated.
exit status 1
Ошибка компиляции для платы LOLIN(WEMOS) D1 R2 & mini.

Пытался из интернета скачать, из папки проекта брал. Ничего.
Также в папке с проектом нет библиотек:
HardwareSerial
ESPmDNS

**Reply by vvip-68** [reply]
Всё это - не отдельные библиотеки, а часть ядра.
ArduinoJson - библиотека, но ее надо брать не какую попало из интернета, а ту, что идет в комплекте с проектом.

**Comment by Inthestep** [comment]
**Памагите.**

**Comment by 33tooth** [comment]
На первой вкладке скетча нужно изменить порядок цветов ленты, ставлю СBGR вместо CRGB, при проверке скетча вылазит ошибка, что не так?

**Reply by vvip-68** [reply]
В каком месте? Откуда вы вообще это взяли? Порядок цветов (определение COLOR_ORDER) обозначается тремя буквами - значение из набора RGB,RBG,GRB,GBR,BRG,BGR
А то что вы написали - это вообще непонятно что.

А CRGB - это вообще имя типа переменной. Оно бывает только такое и никакое другое.

**Reply by 33tooth** [reply]
Значит не туда залез, простите. В каком месте скетча поменять порядок цветов ленты?

**Reply by vvip-68** [reply]
![image](https://github.com/vvip-68/GyverPanelWiFi/assets/39353139/4505cbb2-45de-4340-aea8-27f67e3ddaf3)

**Reply by 33tooth** [reply]
Попробовал найти, вот что выдал поиск:
 // Настройки ленты
  leds =  new CRGB[NUM_LEDS];          
  overlayLEDs = new CRGB[OVERLAY_SIZE];

  // Создать массив для карты индексов адресации светодиодов в ленте
  bool ok = loadIndexMap();
  if (sMATRIX_TYPE == 2 && (!ok || mapListLen == 0)) {
    sMATRIX_TYPE = 0;
    putMatrixSegmentType(sMATRIX_TYPE);
  }


  DEBUGLN("sMATRIX_TYPE=" + String(sMATRIX_TYPE));
  FastLED.addLeds<WS2812, LED_PIN, COLOR_ORDER>(leds, NUM_LEDS).setCorrection( TypicalLEDStrip );

**Reply by vvip-68** [reply]
Ищите еще. Должна быть примерно такая строка:
```
#define COLOR_ORDER       GRB
```
Или у вас не моя версия гирлянды.  Тогда не знаю где там настраивается порядок цветов в ленте. Спрашивайте у автора.

**Reply by 33tooth** [reply]
Скачивал отсюда, GyverPanelWiFi_v1.13, это что другой проект?

**Reply by 33tooth** [reply]
Все, спасибо, нашел!!!
